x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/LengthFunctionProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.expression.function.scalar.string;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.eql.EqlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class LengthFunctionProcessor implements Processor {
+
+    public static final String NAME = "slen";
+
+    private final Processor source;
+
+    public LengthFunctionProcessor(Processor source) {
+        this.source = source;
+    }
+
+    public LengthFunctionProcessor(StreamInput in) throws IOException {
+        source = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(source);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(source.process(input));
+    }
+
+    public static Object doProcess(Object source) {
+        if (source == null) {
+            return null;
+        }
+        if (source instanceof String == false && source instanceof Character == false) {
```

Comment:
I was wondering about this after our discussion today and when thinking of edge cases for `wildcard` https://github.com/elastic/elasticsearch/issues/53999.

What are the conditions that would cause us to hit this path?

Reply:
That's a good question. I cannot think of a scenario where a Character is being received. _Maybe_ the JSON XContent parser would (in a future implementation) consider a single character coming over the wire to be a Character instead of a String.